src/inter-aut.cc

```diff
@@ -0,0 +1,410 @@
+                    opstack.push_back(node);
+                    break;
+                case Mata::FormulaNode::RIGHT_PARENTHESIS:
+                    while (!opstack.back().is_leftpar()) {
+                        output.push_back(opstack.back());
+                        opstack.pop_back();
+                        break;
+                    }
+                    opstack.pop_back();
+                    break;
+                case Mata::FormulaNode::OPERATOR:
+                    for (int j = opstack.size()-1; j >= 0; --j) {
+                        assert(!opstack[j].is_operand());
+                        if (opstack[j].is_leftpar())
+                            break;
+                        if (lower_precedence(node.operator_type, opstack[j].operator_type)) {
+                            output.push_back(opstack[j]);
+                            opstack.erase(opstack.begin()+j);
+                        }
+                    }
+                    opstack.push_back(node);
+                    break;
+                default: assert(false);
+            }
+        }
+
+        while (!opstack.empty()) {
+            output.push_back(opstack.back());
+            opstack.pop_back();
+        }
+
+        return output;
+    }
+
+    /**
+     * Create a graph for a postfix representation of transition formula
+     * @param postfix A postfix representation of transition formula
+     * @return A parsed graph
+     */
+    Mata::FormulaGraph postfix2graph(const std::vector<Mata::FormulaNode> &postfix)
+    {
+        std::vector<Mata::FormulaGraph> opstack;
+
+        for (const auto& node : postfix) {
+            Mata::FormulaGraph gr(node);
+            switch (node.type) {
+                case Mata::FormulaNode::OPERAND:
+                    opstack.push_back(gr);
+                    break;
+                case Mata::FormulaNode::OPERATOR:
+                    switch (node.operator_type) {
+                        case Mata::FormulaNode::NEG:
+                            assert(!opstack.empty());
+                            gr.children.push_back(opstack.back());
+                            opstack.pop_back();
+                            opstack.push_back(gr);
+                            break;
+                        default:
+                            assert(opstack.size() > 1);
+                            gr.children.push_back(opstack.back());
+                            opstack.pop_back();
+                            gr.children.insert(gr.children.begin(), opstack.back());
+                            opstack.pop_back();
+                            opstack.push_back(gr);
+                    }
+                    break;
+                default: assert(false);
+            }
+        }
+
+        assert(opstack.size() == 1);
+
+        return opstack.back();
+    }
+
+    std::vector<Mata::FormulaNode> add_operators_implicitly(const Mata::InterAutomaton& aut,
```

Comment:
I don't get what this function does. It adds `or` operators, but when?

Reply:
yes, it adds `or` operator to initial and final states for NFA where we don't want to make user to write it.